main.tf

```diff
@@ -0,0 +1,141 @@
+resource "aws_s3_bucket" "data" {
```

Comment:
<details><summary>
<img src="https://bc-network-graph-icons.s3.us-west-2.amazonaws.com/prCommentsSeverityIcons/low_new.svg" alt="LOW" width="24px" align="center">&nbsp;&nbsp;<b style="color:black" >Ensure AWS S3 object versioning is enabled</b>
            <br>
            &nbsp;&nbsp;&nbsp;&nbsp;Resource: <a href="https://github.com/marynaKK/terragoat/blob/marynaKK-patch-1/main.tf#L1-L21">aws_s3_bucket.data</a> | Bridgecrew ID: <code>BC_AWS_S3_16</code> | Checkov ID: <a href="https://docs.prismacloud.io/en/enterprise-edition/policy-reference/aws-policies/s3-policies/s3-16-enable-versioning">CKV_AWS_21</a>
            <br>
            
</summary>
<h4>How to Fix</h4>

```tf
+ resource "aws_s3_bucket_versioning" "example" {
+   bucket = aws_s3_bucket.example.id
+ 
+   versioning_configuration {
+     status = "Enabled"
+   }
+ }
```

<h4>Description</h4>


S3 versioning is a managed data backup and recovery service provided by AWS.
When enabled it allows users to retrieve and restore previous versions of their buckets.
S3 versioning can be used for data protection and retention scenarios such as recovering objects that have been accidentally/intentionally deleted or overwritten.

<h4>Benchmarks</h4>
<ul>
<li>FEDRAMP (MODERATE) CP-10, SI-12</li>
<li>PCI-DSS V3.2.1 10.5.3</li>
</ul>

</details>
<p><img src="https://bc-network-graph-icons.s3.us-west-2.amazonaws.com/prCommentsBotLogos/bc.svg" alt="Bridgecrew" width="18px">&nbsp;<small>Comment by Bridgecrew Bot</small></p>